### PR TITLE
Amiga build fix

### PR DESCRIPTION
**Pull request by BSzili**
I managed to switch to SDL2 on MorphOS with some outside help. The same will happen for AmigaOS4 once I'm able to test it myself. There's still no SDL2 for AROS.

### CODE REVIEW

**Comment by Ghabry** [comment]
Needs Rebase. Wrong master used?

SDL2 support is great. We want to drop SDL someday when nobody uses it anymore.

Btw are there some CMake toolchain files available for these platforms? At least with SDL2 support it should be quite easy now to get this working.

File: src/sdl2_ui.cpp [hunk 1]
```diff
@@ -255,8 +255,8 @@ bool Sdl2Ui::RefreshDisplayMode() {
 	int display_height = current_display_mode.height;
 
 #ifdef SUPPORT_ZOOM
-	display_width *= current_display_mode.zoom;
-	display_height *= current_display_mode.zoom;
+		display_width *= current_display_mode.zoom;
```
Comment (carstene1ns) [inline src/sdl2_ui.cpp:258]: rebase leftover.

File: src/sdl2_ui.cpp [hunk 2]
```diff
@@ -411,23 +413,23 @@ void Sdl2Ui::ToggleZoom() {
...
-		current_display_mode.zoom = 1;
-	}
+		// get current window size, calculate next bigger zoom factor
+		int w, h;
+		SDL_GetWindowSize(sdl_window, &w, &h);
+		last_display_mode.zoom = std::min(w / SCREEN_TARGET_WIDTH, h / SCREEN_TARGET_HEIGHT);
+		current_display_mode.zoom = last_display_mode.zoom + 1;
+
+		// get maximum usable window size
+		int display_index = SDL_GetWindowDisplayIndex(sdl_window);
```
Comment (carstene1ns) [inline src/sdl2_ui.cpp:423]: same here.

**Comment by BSzili** [comment]
CMake is now part of the official MorphOS SDK, there's also a third party port to AmigaOS4. AROS again falls by the wayside, but I cross-compile everything. When you decide to drop SDL1.2, I'll try to come up with something. Unless you plan to use multiple windows it should be possible to make an SDL2 shim for SDL1.2 on AROS.

**Comment by carstene1ns** [comment]
We will likely have some SDL1.2 support for the next years. However, only fixes, no new features.

**Comment by fmatthew5876** [comment]
Unrelated to the PR, @BSzili what gcc compiler versions are supported for these platforms? Is there work being done on upgrading?

I'd like to get us on C++17 as soon as all our ports are on a viable gcc version.

**Comment by BSzili** [comment]
MorphOS has GCC 9, and AmigaOS4 has GCC 8, those should be good. For AROS ABIv0 the last compiler is GCC 6, which only has limited C++17 support :\

**Comment by carstene1ns** [comment]
Also pulling this into maintenance branch.

**Pull request merged**